src/tester.rs

```diff
@@ -422,42 +425,62 @@ fn run_tests(
                     None => continue,
                 };
                 let mut meant_to_error = false;
+                // First, see which of the tests passed.
                 if let Some(ref status) = test.status {
-                    match status {
-                        Status::Success => {
-                            if !output.status.success() {
-                                failure.status = Some("Error".to_owned());
-                            }
-                        }
+                    pass_status = Some(match status {
+                        Status::Success => output.status.success(),
                         Status::Error => {
                             meant_to_error = true;
-                            if output.status.success() {
-                                failure.status = Some("Success".to_owned());
-                            }
+                            !output.status.success()
                         }
-                        Status::Int(i) => {
-                            let code = output.status.code();
-                            if code != Some(*i) {
+                        Status::Int(i) => output.status.code() == Some(*i),
+                    });
+                }
+                let stderr_utf8 = String::from_utf8(output.stderr).unwrap();
+                if let Some(ref stderr) = test.stderr {
+                    pass_stderr = Some(fuzzy::match_vec(stderr, &stderr_utf8));
+                }
+                let stdout_utf8 = String::from_utf8(output.stdout).unwrap();
+                if let Some(ref stdout) = test.stdout {
+                    pass_stdout = Some(fuzzy::match_vec(stdout, &stdout_utf8));
+                }
+
+                // If something failed, we want to print out everything which either failed a test
+                // or wasn't part of a test.
```

Comment:
I'm not sure what you mean by printing stuff which isn't part of a test here.

Reply:
Does the commit message make this clearer?

Reply:
I think I might have it. It prints anything that wasn't matched?

Reply:
Correct.